### ISSUE TITLE
Load Pdf file from byte array/stream

### ISSUE THREAD

**Issue by MasterIPI** [QuestPDF/QuestPDF#899]
Just a feature request/question

Is there any plans on creating pdf "Document" from a byte array/stream of existing pdf file?

It would be great if we could get already existing pdf files as byte arrays/streams and then merge with the generated file or just merge existing pdf files.

**Comment by rima1098** [comment]
Have a look at https://www.pdfsharp.net/. It is possible to merge PDFs with this tool and might be sufficient for your needs

**Comment by HakamFostok** [comment]
`Document.Merge` would really be helpful if we could merge already existing files in the Disk.
Right now merging documents will work for documents created by the code only.
Hope to see this feature.

**Comment by MasterIPI** [comment]
> Have a look at https://www.pdfsharp.net/. It is possible to merge PDFs with this tool and might be sufficient for your needs

I know about this library and currently we are using IronPdf, but we have a request to move from that to something else, to find out if some other library works faster that IronPdf and without drops in performance if run within docker linux container.

**Comment by HakamFostok** [comment]
No, no, my situation was much simpler, I just wanted to combine a couple of pdfs files in one, locally on my machine, that is it.
I switched to pdfSharp and it solved my problem completely.
Thanks a lot

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this issue due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

The functionality you requested is now available through the `DocumentOperation` API, introduced in version 2024.12.0. You can load existing PDF files from disk and merge them together. Full documentation is available here:
https://www.questpdf.com/concepts/document-operations.html

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
